catalogue_of_life/Loader.java

```diff
@@ -0,0 +1,242 @@
+                if (zipEntry.isDirectory()) {
+                    Files.createDirectories(resolvedPath);
+                } else {
+                    Files.createDirectories(resolvedPath.getParent());
+                    Files.copy(zipIn, resolvedPath, StandardCopyOption.REPLACE_EXISTING);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void prepareData(Path dataDirectory) throws IOException {
+        TsvParser distributionsParser = beginParsingFile(dataDirectory.resolve("Distribution.tsv"));
+        TsvWriter marineRegionsWriter = createTsvWriter(dataDirectory.resolve("MarineDistribution.tsv"));
+        TsvWriter describedRegionsWriter = createTsvWriter(dataDirectory.resolve("DescribedDistribution.tsv"));
+
+        Record record;
+        Map<String, String> recordValues = new HashMap<>();
+        while ((record = distributionsParser.parseNextRecord()) != null) {
+            String gazetteer = record.getString("col:gazetteer");
+            if (gazetteer.equals("text") || gazetteer.equals("iso")) {
+                record.fillFieldMap(recordValues);
+                describedRegionsWriter.writeRow(recordValues);
+            } else if (gazetteer.equals("mrgid") && isValidMarineRegionID(record.getString("col:areaID"))) {
+                record.fillFieldMap(recordValues);
+                marineRegionsWriter.writeRow(recordValues);
+            }
+        }
+
+        marineRegionsWriter.close();
+        describedRegionsWriter.close();
+    }
+
+    private static TsvParser beginParsingFile(Path tsvFile) throws IOException {
+        TsvParserSettings settings = new TsvParserSettings();
+        settings.setHeaderExtractionEnabled(true);
+        settings.setMaxCharsPerColumn(1_000_000);
+
+        TsvParser parser = new TsvParser(settings);
+        InputStream inputStream = Files.newInputStream(tsvFile);
+        parser.beginParsing(inputStream);
+
+        return parser;
+    }
+
+    private static TsvWriter createTsvWriter(Path tsvFile) throws IOException {
+        TsvWriterSettings writerSettings = new TsvWriterSettings();
+        writerSettings.setHeaderWritingEnabled(true);
+        writerSettings.setMaxCharsPerColumn(1_000_000);
+
+        return new TsvWriter(Files.newOutputStream(tsvFile), writerSettings);
+    }
+
+    private static boolean isValidMarineRegionID(String areaID) {
+        return isNumeric(areaID);
+    }
+
+    private static boolean isNumeric(String s) {
+        try {
+            Long.parseLong(s);
+        } catch (NumberFormatException e) {
+            return false;
+        }
+        return true;
+    }
+
+    public static void loadData(Path dataDirectory, Options options) throws IOException {
+        extractSchema(dataDirectory);
+
+
+        LoadOptions loadOptions = new LoadOptions();
+        loadOptions.dataConfigFilePath = createTypeDBLoaderConfig(dataDirectory).toString();
```

Comment:
oh dear this should take a `Path` not a `String` haha, mind filing an issue in Loader?

Reply:
It's a picocli options handler, which cannot really take a Path from the command line without some serious boilerplate. String is fine.